src/accordion.ts

```diff
@@ -0,0 +1,188 @@
+import { LitElement, html } from "lit";
+import {
+    customElement,
+    property,
+    queryAssignedElements,
+} from "lit/decorators.js";
+
+@customElement("aalam-accordion")
+export class AalamAccordion extends LitElement {
+    @property()
+    togglesel: string = ".acc-title";
+
+    @property()
+    animationDur: number = 200;
+
+    @property()
+    bodysel: string = ".acc-body";
+
+    @property({ type: Boolean })
+    nocloseall: boolean = false;
+
+    @property({ type: Boolean })
+    nomultiple: boolean = false;
+
+    @property({ type: String })
+    activecls: string = "acc-active";
+
+    private openItems: Element[] = [];
+
+    @queryAssignedElements({ flatten: true })
+    private items!: Element[];
+
+    private handleNomultipleChange() {
+        if (this.nomultiple && this.openItems.length > 1) {
+            this.openItems.slice(1).forEach((item) => {
+                this.closeItem(item);
+            });
+        }
+    }
+
+    private handleNocloseallChange() {
+        if (
+            this.nocloseall &&
+            this.openItems.length === 0 &&
+            this.items.length > 0
+        ) {
+            this.openItem(this.items[0]);
+        }
+    }
+
+    override render() {
+        return html`<slot @slotchange=${this.handleSlotChange}
+                          @transitionend=${this._handleTransitionEnd}></slot>`;
+    }
+    private _handleTransitionEnd(event: any) {
+        const body = event.target;
+        if (body.style.height == "0px") {
+            body.style.display = "none";
+        }
+        body.style.height = "";
+        body.style.overflow = "";
+        body.style.transitionProperty = "";
+        body.style.transitionDuration = "";
+        body.style.transitionTimingFunction = "";
+    }
+
+    override attributeChangedCallback(
+        name: string,
+        _old: string | null,
+        value: string | null
+    ): void {
+        super.attributeChangedCallback(name, _old, value);
+        if (name === "nomultiple") {
+            this.handleNomultipleChange();
+        } else if (name === "nocloseall") {
+            this.handleNocloseallChange();
+        }
+    }
+
+    private handleSlotChange() {
+        this.openItems = [];
+        this.items.forEach((item, index) => {
+            const toggleElement = item.querySelector(this.togglesel);
+            if (toggleElement) {
+                toggleElement.addEventListener("click", () =>
+                    this.toggle(index)
+                );
+            }
+
+            if (item.classList.contains(this.activecls)) {
+                this.openItems.push(item);
+                this.openItem(item, false);
+            } else {
+                const body = item.querySelector(this.bodysel) as HTMLElement;
+                if (body) {
+                    body.style.display = "none";
+                }
+            }
+        });
+
+        this.handleNomultipleChange();
+        this.handleNocloseallChange();
```

Comment:
We dont need this as the closeAll and noMultiple will be handled then and there in the open item and close item call.